plugins/module_utils/constants.py

```diff
@@ -0,0 +1,129 @@
+    "contract": {"match_value": "contractMatch", "pattern_value": "contract", "permitted_objects": ["tenant", "contract"]},  # currently not used
+}
+
+ETHER_TYPES = ["arp", "fcoe", "ip", "mac_security", "mpls_unicast", "trill"]
+
+PROTOCOL_TYPES = ["all", "egp", "eigrp", "icmp", "icmpv6", "igmp", "igp", "l2tp", "ospfigp", "pim", "tcp", "udp"]
+
+OPERATORS = ["contains", "begins_with", "ends_with", "equal_to", "not_equal_to", "not_contains", "not_begins_with", "not_ends_with"]
+
+CONFIG_OPERATORS = ["regex", "exact", "at_least", "at_most", "all", "none", "at_least_one"]
+
+PARAMETER_TYPES = {
+    "name": {"parameter_value": "CCP_NAME", "valid_for": ["epg", "bd", "vrf", "contract", "subject", "filter"]},  # currently not used
+    "name_alias": {"parameter_value": "CCP_NAME_ALIAS", "valid_for": ["epg", "bd", "vrf", "contract", "subject", "filter"]},  # currently not used
+    "enforcement_preference": {
+        "parameter_value": "CCP_ENFORCEMENT_PREFERENCE",
+        "valid_for": ["vrf", "epg"],  # currently not used
+        "permitted_values": ["Unenforced", "Enforced"],  # currently not used
+    },
+    "enforcement_direction": {
+        "parameter_value": "CCP_ENFORCEMENT_DIRECTION",
+        "valid_for": ["vrf"],  # currently not used
+        "permitted_values": ["Ingress", "Egress"],  # currently not used
+    },
+    "preferred_group": {
+        "parameter_value": "CCP_PREFERRED_GROUP",
+        "valid_for": ["vrf"],  # currently not used
+        "permitted_values": ["Disabled", "Enabled"],  # currently not used
+    },
+    "bd_enforcement": {
+        "parameter_value": "CCP_BD_ENFORCEMENT",
+        "valid_for": ["vrf"],  # currently not used
+        "permitted_values": ["No", "Yes"],  # currently not used
+    },
+    "bd_type": {"parameter_value": "CCP_BD_TYPE", "valid_for": ["bd"], "permitted_values": ["Regular", "Fc"]},  # currently not used  # currently not used
+    "l2_unknown_unicast": {
+        "parameter_value": "CCP_L2_UNKNOWN_UNICAST",
+        "valid_for": ["bd"],  # currently not used
+        "permitted_values": ["Hardware Proxy", "Flood"],  # currently not used
+    },
+    "l3_unknown_unicast_flooding": {
+        "parameter_value": "CCP_L3_UNKNOWN_MULTICAST_FLOODING",
+        "valid_for": ["bd"],  # currently not used
+        "permitted_values": ["Optimized Flood", "Flood"],  # currently not used
+    },
+    "bd_multi_destination_flooding": {
+        "parameter_value": "CCP_BD_MULTI_DESTINATION_FLOODING",
+        "valid_for": ["bd"],  # currently not used
+        "permitted_values": ["drop", "bd-flood", "encap-flood"],  # currently not used
+    },
+    "pim": {"parameter_value": "CCP_PIM", "valid_for": ["bd"], "permitted_values": ["Disabled", "Enabled"]},  # currently not used  # currently not used
+    "arp_flooding": {
+        "parameter_value": "CCP_ARP_FLOODING",
+        "valid_for": ["bd"],  # currently not used
+        "permitted_values": ["No", "Yes"],  # currently not used
```

Comment:
Any yes/no if on boolean could be change it to true/false?

Reply:
I agree with your comment but in this instance it is a representation of what is permitted as input for the paramater on NDI side. Which is not true/false but Yes/No. 

For some additional context, I created this overview during the phase of understanding the different configuration possibilities. I kept this in for informational purposes ( not wasting the effort of getting all the possible values ) only thus also the comment `# currently not used` is put behind.